Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/InteropLibraryCreation.kt

```diff
@@ -27,9 +31,16 @@ fun createInteropLibrary(arguments: LibraryCreationArguments) {
             abiVersion = KotlinAbiVersion.CURRENT,
             compilerVersion = KonanVersion.CURRENT
     )
+    val outputPathWithoutExtension = arguments.outputPath.removeSuffixIfPresent(".klib")
     KonanLibraryWriterImpl(
-            File(arguments.outputPath),arguments.moduleName, version, arguments.target
+            File(outputPathWithoutExtension),
+            arguments.moduleName,
+            version,
+            arguments.target
     ).apply {
+        // TODO: Add write strategy that splits big fragments.
```

Comment:
Does `kotlinx.metadata` perform splitting?

Reply:
[`KlibModuleMetadata.write`](https://github.com/JetBrains/kotlin/blob/4438dd282fa15b3afca32096bf81a75e8a9967e6/libraries/kotlinx-metadata/klib/src/kotlinx/metadata/klib/KlibModuleMetadata.kt#L103) has a parameter that allows to override default writing [strategy](https://github.com/JetBrains/kotlin/blob/4438dd282fa15b3afca32096bf81a75e8a9967e6/libraries/kotlinx-metadata/klib/src/kotlinx/metadata/klib/KlibModuleMetadata.kt#L38). I will implement splitting strategy in the following PR.